Unified pointing and pol_angle computation interface

anand-avinash opened the pull request:
This PR addresses the discussions made in #376. The aim of this PR is to remove the code redundancies in computing the pointings and polarization angles at different places in the framework.

ziotom78 commented on test/test_detector_blocks.py:
```diff
@@ -79,7 +79,8 @@ def test_detector_blocks(dets=dets, sampling_freq_Hz=sampling_freq_Hz):
 
         # No testing required if the proc doesn't owns a detector
-        if obs.det_idx is not None:
+        print(f"rank = {comm.rank}, {obs.det_idx}, {[idx for idx in obs.det_idx]}")
```
Is this `print` necessary?

anand-avinash replied:
No, I just forgot to remove it!

ziotom78 reviewed (commented):
This looks great! So far, it looks fine.

One small question: a few error messages associated with `assert` statements have disappeared; what does the code do now if those conditions are false? (I tried to follow the code paths but quickly got lost…)

anand-avinash commented:
> One small question: a few error messages associated with assert statements have disappeared; what does the code do now if those conditions are false? (I tried to follow the code paths but quickly got lost…)

Do you mean these errors: <https://github.com/litebird/litebird_sim/actions/runs/14198954601/job/39780913438>?

If yes, I changed the way the default values of non-numeric detector parameters are assigned in the Observation class. Instead of setting them to `None`, I changed them to `[None]`, since the detector information need to be a list or array. To address the error messages, I simply replaced `assert obs.row_int is None` with `assert obs.row_int == [None]` and same wherever applicable.

mj-gomes commented:
> To address the error messages, I simply replaced `assert obs.row_int is None` with `assert obs.row_int == [None]` and same wherever applicable.


Hi! Those changes are already in the main branch (they were included in one of my last PR). If you rebase this branch, the problem should go away.

ziotom78 commented:
> > One small question: a few error messages associated with assert statements have disappeared; what does the code do now if those conditions are false? (I tried to follow the code paths but quickly got lost…)
> 
> Do you mean these errors: https://github.com/litebird/litebird_sim/actions/runs/14198954601/job/39780913438?

No, sorry, I was referring to a few `log.warning()` statements, but now I think I have found them in the new code, so ok!

ziotom78 commented:
I removed an unused import that caused the tests to fail. @anand-avinash, can this PR be merged now? We will then need to address issue #376 by adding the HWP angle to ψ, but I would like to open a new PR for clarity.

martamonelli commented:
@paganol and I just went through the changes and everything looks good to us.

@anand-avinash, if you're happy with it, it would be great if you could merge the PR, so we can open another branch from master and start working on the interface with sanepic

anand-avinash commented:
Thank you for going through it. I will merge it right away.
